[DAL] Extend algorithm anatomy (#212)
oneapi-src/oneAPI-spec: source/elements/oneDAL/source/programming_model/algorithm_anatomy.rst

   } // namespace oneapi::dal::%ALGORITHM%

.. _tasks:

Computational Tasks
-------------------
The supported computational tasks are declared within the ``%ALGORITHM%::task``
namespace using tag-types. Algorithm shall support at least one task and declare

> the ``by_default`` type alias that refers to one of the tasks as shown in the

Reviewer: Somewhere, in the section or earlier we need a sentence that explains specifics of the task in cases such as covariance

example below.

> If an algorithm assumes separation on ``classification`` and ``regression`` tasks,

Reviewer 2: "assumes separation on" is not a correct structure 

```suggestion
If an algorithm assumes both``classification`` and ``regression`` tasks,
```
